src/form-engine.test.tsx

```diff
@@ -842,15 +844,68 @@ describe('Form engine component', () => {
     });
   });
 
-  function renderForm(formUUID, formJson, intent?: string) {
+  describe('Read only mode', () => {
+    it('should ascertain that each field with readonly = true passed will not be editable', async () => {
+      await act(async () => {
+        renderForm(null, readOnlyValidationForm);
+      });
+
+      const visitTypeDropdown = screen.getByRole('combobox', {
+        name: /visit type/i,
+      });
+      const visitPunctualityTextbox = screen.getByLabelText(/visit punctuality/i);
+
+      expect(visitTypeDropdown).toBeInTheDocument();
+      expect(visitTypeDropdown).toHaveClass('cds--list-box__field');
+
+      const visitTypeWrapper = visitTypeDropdown.closest('.cds--dropdown');
+      expect(visitTypeWrapper).toHaveClass('cds--dropdown cds--dropdown--readonly cds--list-box');
+
+      expect(visitPunctualityTextbox).toBeInTheDocument();
+      expect(visitPunctualityTextbox).toHaveClass('cds--text-input');
+
+      const visitPunctualityWrapper = visitPunctualityTextbox.closest('.cds--text-input-wrapper');
+      expect(visitPunctualityWrapper).toHaveClass(
+        'cds--form-item cds--text-input-wrapper cds--text-input-wrapper--readonly',
+      );
+    });
+  });
+
+  describe('Form view mode', () => {
+    it('should ascertain that fields cannot be edited on view mode', async () => {
+      await act(async () => {
+        renderForm(null, htsPocForm, null, 'view');
+      });
+      const testingHistoryButton = screen.getByRole('button', { name: /Testing history/i });
+      expect(testingHistoryButton).toBeInTheDocument();
+
+      const hivTestButton = screen.getByRole('button', { name: /When was the HIV test conducted\?:/i });
+      expect(hivTestButton).toBeInTheDocument();
+
+      const blankFields = screen.getAllByText(/\(Blank\)/i);
+      blankFields.forEach((blankField) => {
+        expect(blankField).toBeInTheDocument();
+      });
+
+      const inputs = screen.queryAllByRole('textbox');
+      inputs.forEach((input) => {
+        expect(input).toHaveAttribute('readonly');
+      });
```

Comment:
Can you also assert that the "Save" button is disabled?

Reply:
Added it on line 884